src/librustc_resolve/resolve_imports.rs

```diff
@@ -887,14 +887,24 @@ impl<'a, 'b:'a> ImportResolver<'a, 'b> {
         // Since import resolution is finished, globs will not define any more names.
         *module.globs.borrow_mut() = Vec::new();
 
+        let report_already_exported = |this: &mut Self, ident, span, prev_span| {
+            let msg = format!("a macro named `{}` has already been exported", ident);
+            this.session.struct_span_err(span, &msg)
```

Comment:
I'm ok with a hard error, but it seems like there *is* a defined semantics, right? (Shadowing) We could always issue warnings, in other words.